### PR TITLE
Closing fds

### PR DESCRIPTION
**Pull request by jreidinger**
I tested two approaches. The first one was trivial iterative approach with some reasonable max and try to close everything. The second one uses proc filesystem and gets info from it how it behave.

Tested with this script:
```ruby
require 'benchmark'

Benchmark.bm(20) do |bm|
  20.times { File.open("/dev/zero") }
  bm.report("Closing all fds:") do
    3.upto(1024) do |i|
      begin
        IO.new(i).close
      rescue ArgumentError, Errno::EBADF
      end
    end
  end

  20.times { File.open("/dev/zero") }
  bm.report("Using self proc fd") do
    Dir.glob("/proc/self/fd/*").each do |path|
      fd = File.basename(path).to_i
      next if (0..2).include?(fd)

      begin
        IO.new(fd).close
      rescue ArgumentError, Errno::EBADF
      end
    end
  end
end
```

and result shows that second approach is better:

```
                           user     system      total        real
Closing all fds:       0.001745   0.000000   0.001745 (  0.001744)
Using self proc fd     0.000152   0.000000   0.000152 (  0.000152)
```

Also it reveal that time spend is very small, so overhead should be acceptable.

Related bug report we get about leaking vgs https://bugzilla.suse.com/show_bug.cgi?id=1151960 that is probably caused by this.

### CODE REVIEW

**Comment by jreidinger** [comment]
Manual testing is still needed to do. Also it would be nice to enable travis and update rubocop.

**Comment by lslezak** [comment]
Um, I do not the admin permissions for this GitHub repo so I cannot enable Travis for it. If you do have the permissions enable it at https://travis-ci.org/organizations/openSUSE/repositories.

**Comment by jreidinger** [comment]
travis enabled

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,8 @@
+0.5.1 (2019-10-16)
+------------------
+
+* Implement closing open fds after call to fork
```
Comment (mvidner) [inline CHANGELOG]: Please mention the bug here, https://bugzilla.suse.com/show_bug.cgi?id=1151960

**Comment by jreidinger** [comment]
manual testing proof that it really fixes https://bugzilla.suse.com/show_bug.cgi?id=1151960
with 0.5.0 leaked fds in logs. With 0.5.1 no such entries.

File: .rubocop.yml [hunk 2]
```diff
@@ -49,3 +54,7 @@ Style/StringLiterals:
 # Configuration parameters: WordRegex.
 Style/WordArray:
   Enabled: false
+
+# we intentionally ignore some exceptions in fork as we have no way to report it back
```
Comment (mvidner) [inline .rubocop.yml]: NP: This applies to all exceptions. Better disable the cop in the specific place where we've checked it is safe.

**Pull request merged**